### PR TITLE
moved all occurrences of RE_USERNAME to src/utils/regex.rs

### PR DESCRIPTION
**Pull request by meguminloli**

### CODE REVIEW

File: src/routes/servers/server_create.rs [hunk 1]
```diff
@@ -77,6 +77,8 @@ pub async fn req(user: User, info: Json<Data>) -> Result<Value> {
 
         icon: None,
         banner: None,
+
+        flags: None,
```
Comment (yannickfunk) [inline src/routes/servers/server_create.rs:81]: what's the point of this addition?

File: src/util/regex.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+pub static RE_USERNAME: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[a-zA-Z0-9_.]+$").unwrap());
```
Comment (yannickfunk) [inline src/util/regex.rs]: Add a newline at the end of the file

**Review by yannickfunk — CHANGES_REQUESTED** [review]

File: src/routes/servers/server_create.rs (hunk 1 above)
Comment (insertish) [reply src/routes/servers/server_create.rs:81]: Wasn't this fixed in b4db321d0ef1ff296f6a611330cc1db73e5e566e already?

Comment (sneedcat) [reply src/routes/servers/server_create.rs:81]: Yeah it was, I cloned from an older commit.

**Review by insertish — APPROVED** [review]

**Pull request merged**